### PR TITLE
GEN-827 | Partner Init: pre-fill information from search params in price intent

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Pre-fill information passed as query params to the initial price intent

- Refactor API route to not make it too complicated

## Justify why they are needed

- Mimic how partner widget works

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2880** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2880" target="_blank">Graphite</a> 
    * **PR #2882** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2882" target="_blank">Graphite</a> 
      * **PR #2883** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883" target="_blank">Graphite</a>  👈
        * **PR #2884** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2884" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883?utm_source=stack-comment).

File: apps/store/src/services/partner/parseSearchParams.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import {
+  PartnerWidgetInitInput,
+  ShopSessionCustomerUpdateInput,
+  TypeOfContract,
+} from '@/services/apollo/generated'
+import { getCountryByLocale } from '@/utils/l10n/countryUtils'
+import { isIsoLocale, toRoutingLocale } from '@/utils/l10n/localeUtils'
+import { IsoLocale, RoutingLocale } from '@/utils/l10n/types'
+
+const DEFAULT_LOCALE: IsoLocale = 'sv-SE'
```
Comment (alebedev) [inline apps/store/src/services/partner/parseSearchParams.ts:11]: Can we reuse common settings and get ISO locale from it?

File: apps/store/src/services/partner/parseSearchParams.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    ...(externalRequestId && { externalRequestId }),
+  }
+
+  const searchSsn = searchParams.get(SearchParam.Ssn)
+  let ssn: string | undefined = undefined
+  if (searchSsn && Personnummer.valid(searchSsn)) {
+    ssn = Personnummer.parse(searchSsn).format(true)
+  }
+  const email = searchParams.get(SearchParam.Email)
+  const firstName = searchParams.get(SearchParam.FirstName)
```
Comment (alebedev) [inline apps/store/src/services/partner/parseSearchParams.ts:77]: Should we drop first and last name?

**Review by alebedev — APPROVED** [review]
Looks good!

File: apps/store/src/services/partner/parseSearchParams.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/services/partner/parseSearchParams.ts:77]: Hmm probably we should -- there are no inputs for them

File: apps/store/src/pages/api/partner/init.ts [hunk 3]
```diff
@@ -1,118 +1,52 @@
...
+
+    const apolloClient = await initializeApolloServerSide({ locale, req, res })
+    const { id: shopSessionId, partnerName } = await createPartnerShopSession(
+      apolloClient,
+      partnerWidgetInitVariables,
+    )
+    setupShopSessionServiceServerSide({ apolloClient, req, res }).saveId(shopSessionId)
+    console.info(`Partner Init | Shop Session = ${shopSessionId}, Partner = ${partnerName}`)
+
+    const service = priceIntentServiceInitServerSide({ req, res, apolloClient })
```
Comment (guilhermespopolin) [inline apps/store/src/pages/api/partner/init.ts:29]: nit:

```suggestion
    const priceIntentService = priceIntentServiceInitServerSide({ req, res, apolloClient })
```

**Review by guilhermespopolin — APPROVED** [review]
Nice refactoring with the search parameters parser.

Added via Giphy

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883).

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883) rebased this pull request as part of a merge.

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2883).

**Pull request merged**